app/src/main/java/com/example/drawingApp/MainActivity.kt

```diff
@@ -2,101 +2,121 @@ package com.example.drawingApp
 
 import android.os.Bundle
 import android.os.StrictMode
-import android.view.LayoutInflater
-import android.view.View
-import android.widget.EditText
+import android.util.Log
 import android.widget.TextView
-import android.widget.Toast
 import androidx.activity.viewModels
 import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
-import androidx.core.view.isVisible
-import androidx.core.widget.addTextChangedListener
+
 
 class MainActivity : AppCompatActivity() {
     private val model: DrawingViewModel by viewModels()
+    private var isAlertDialogOpen = false
+    private var isLayerSheetOpen = false
+
+    var botSheetObj: DialogUtility.SheetObject? = null
+    var alertDialog: AlertDialog? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
         enableStrictMode()
         val colorCircleView = findViewById<SelectedColorView>(R.id.colorCircle)
         val drawingFieldView = findViewById<DrawingFieldView>(R.id.drawField)
+        var onSubmission: (String?) -> Boolean
 
         drawingFieldView.onBitmapUpdate = {
             model.activeBitmap = it
         }
 
-        findViewById<TextView>(R.id.newLayer).setOnClickListener {
-            showLayerAlertDialog()
-        }
-
-        model.activeBitmap?.let {
-            drawingFieldView.setBitmap(it)
-            drawingFieldView.invalidate()
-        }
-
         colorCircleView.onColorChange = { stateColor ->
             model.primaryColor = stateColor
             drawingFieldView.updateColor(stateColor)
         }
+        //whenever we recieve a need to update the state of the view
+        model.onUpdate = { state ->
+            var creating = true
 
-        if (model.isLayerCreationDialogOpen) showLayerAlertDialog()
-
-    }
-
-    fun showLayerAlertDialog() {
-        val inflater = LayoutInflater.from(this@MainActivity)
-        val view = inflater.inflate(R.layout.layer_dialog, null)
-        val layerField = view.findViewById<EditText>(R.id.newLayerField)
-        val errorText = view.findViewById<TextView>(R.id.dialogError)
-        var layerCounter = 1
+            state.activeBitmap?.let {
+                drawingFieldView.setBitmap(it)
+                drawingFieldView.invalidate()
+            }
+            //based on the state of the sheet alertdialog being open or closed as well as if the
+            //sheet is open or closed, change how the submission function of the dialog works
+            if (state.isLayerSheetOpen && state.isAlertDialogOpen) {
+                creating = false
+                onSubmission = {
```

Comment:
You have enough overlap in these `onSubmissions` you should combine them

Reply:
combined